### PR TITLE
Add: Support let user add global filter for go-restful server

### PR DESCRIPTION
**Pull request by Patrick0308**
**What this PR does**:
Let user can add global Filter, for example
```
server_impl.AddGoRestfulServerFilter(func(request *restful.Request, response *restful.Response, chain *restful.FilterChain) {
	println(request.SelectedRoutePath())
	chain.ProcessFilter(request, response)
})
server_impl.AddGoRestfulServerFilter(func(request *restful.Request, response *restful.Response, chain *restful.FilterChain) {
	println("filter2")
	chain.ProcessFilter(request, response)
})
config.Load()
```
Take notice that you should add filter before config.Load()

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: protocol/rest/server/server_impl/go_restful_server.go [hunk 1]
```diff
@@ -59,6 +61,11 @@ func NewGoRestfulServer() *GoRestfulServer {
 
 func (grs *GoRestfulServer) Start(url common.URL) {
 	grs.container = restful.NewContainer()
+	if len(filterSlice) > 0 {
+		for _, filter := range filterSlice {
```
Comment (hxmhlt) [inline protocol/rest/server/server_impl/go_restful_server.go]: is `if len(filterSlice) > 0`  need ?

File: protocol/rest/server/server_impl/go_restful_server.go [hunk 2]
```diff
@@ -20,6 +20,7 @@ package server_impl
 import (
 	"context"
 	"fmt"
+	"github.com/apache/dubbo-go/protocol/rest/server"
```
Comment (zouyx) [inline protocol/rest/server/server_impl/go_restful_server.go]: split package

**Comment by zouyx** [comment]
i would like to know what is the usage scenarios for this PR

**Comment by AlexStocks** [comment]
LGTM

**Pull request merged**